fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/temp/CascadesPlanner.java

```diff
@@ -229,7 +230,7 @@ public RecordQueryPlan plan(@Nonnull RecordQuery query) {
                         "explain", PlannerGraphProperty.explain(singleRoot)));
             }
 
-            return (RecordQueryPlan)singleRoot;
+            return new QueryPlanResult((RecordQueryPlan)singleRoot);
```

Comment:
can we add the task queue size metrics here so this PR serves an immediate purpose?

Reply:
Done.